noether_gui/include/noether_gui/widgets/configurable_tpp_pipeline_widget.h

```diff
@@ -21,7 +21,9 @@ class TPPPipelineWidget;
 class ConfigurableTPPPipelineWidget : public QWidget
 {
 public:
-  ConfigurableTPPPipelineWidget(boost_plugin_loader::PluginLoader loader, QWidget* parent = nullptr);
+  ConfigurableTPPPipelineWidget(boost_plugin_loader::PluginLoader loader,
+                                QWidget* parent = nullptr,
+                                std::string default_configuration_file_directory = "");
```

Comment:
One little nitpick: let's move the QWidget parent argument to be the last in the list. I think the configuration file directory argument will be set more frequently than the parent widget argument. 

```suggestion
  ConfigurableTPPPipelineWidget(boost_plugin_loader::PluginLoader loader,
                                std::string default_configuration_file_directory = "",
                                QWidget* parent = nullptr);
                                
```